### DISCUSSION TITLE
How to use multiple colors in one text

### DISCUSSION THREAD

**Discussion by Farman52** [samizdatco/skia-canvas#110] [Q&A] [answered]
I would like to ask how could I possibly use multiple colors in just one text
```js
ctx.font = '30px Arial'
ctx.fillStyle = "#fcba03"
ctx.textAlign = 'center'
ctx.fillText('[MVP+] Farmans', (790-150)/2+150, 43)
```
![image](https://user-images.githubusercontent.com/32415385/183264756-b7a2ac8b-a4a1-4e11-9863-cce21f159a1b.png)
Is it possible to have different color for the plus in the prefix? [MVP+], for example blue

**Comment by samizdatco** [comment, marked as answer]
It's not possible in a single step. You need to use a combination of `fillText` and `measureText` to lay out the line yourself and change the `fillStyle` in between. 

Something like this will work:
```js
let prefix = "correct horse ",
    highlighted = "battery",
    suffix = " staple"

let x = 100,
    y = 100

ctx.font = '20px Helvetica'
ctx.fillText(prefix, x, y)
x += ctx.measureText(prefix).width

ctx.fillStyle = 'red'
ctx.fillText(highlighted, x, y)
x += ctx.measureText(highlighted).width

ctx.fillStyle = 'black'
ctx.fillText(suffix, x, y)
```
<img width="309" alt="image" src="https://user-images.githubusercontent.com/469523/183302873-0c9a5d32-9960-4501-a062-aaea48aaf387.png">

**Reply by Farman52** [reply]
Thanks, looks so simple now
